External/INCHI-API/inchi.cpp

```diff
@@ -1246,18 +1246,32 @@ void cleanUp(RWMol &mol) {
         break;
 
     }  // end the switch block
-  }  // end the for loop that iterates over atoms
+  }    // end the for loop that iterates over atoms
 }  // end cleanUp
+
+class ManagedCharArray {
+ public:
+  ManagedCharArray() = delete;
+  ManagedCharArray(const std::string &str) : m_data(new char[str.size() + 1]) {
+    strcpy(m_data, str.c_str());
+  }
+  ~ManagedCharArray() { delete[] m_data; }
+  char *get() { return m_data; }
+
+ private:
+  char *m_data = nullptr;
+};
+
 }  // namespace
 
 RWMol *InchiToMol(const std::string &inchi, ExtraInchiReturnValues &rv,
                   bool sanitize, bool removeHs) {
   // input
-  char *_inchi = new char[inchi.size() + 1];
+  ManagedCharArray _inchi(inchi);
```

Comment:
I would just use

```
std::vector<char> _inchi(inchi.begin(), inchi.end()); 
```

As opposed to rolling a new class.

Reply:
Or you can use alloca and assume the stack isn't going to blow up.

Reply:
... of course, it's never that easy, and we need a null terminator :P